### PR TITLE
LPS-165565 Make sure the object formula field should not be used as an object aggregation filter

### PR DESCRIPTION
**Pull request by seltong**

### CODE REVIEW

**Comment by seltong** [comment]
ci:test:relevant

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java [hunk 1]
```diff
@@ -1822,6 +1830,43 @@ private GroupByStep _getOneToManyObjectEntriesGroupByStep(
...
+		if (StringUtil.equals(
+				outputType, ObjectFieldConstants.BUSINESS_TYPE_BOOLEAN)) {
+
+			return GetterUtil.getBoolean(value);
+		}
+
+		if (StringUtil.equals(
+				outputType, ObjectFieldConstants.BUSINESS_TYPE_DATE)) {
+
+			DateFormat dateFormat = DateFormatFactoryUtil.getSimpleDateFormat(
```
Comment (marcelabc) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java:1843]: Hey @seltong,
I think we should format considering the user locale. I search a little and found this:
`DateFormat dateFormat = DateFormatFactoryUtil.getDate(locale);`
You can search for other things...
What do you think? Could you please test?

**Review by marcelabc — CHANGES_REQUESTED** [review]
Hi 
I finished my review. Let me know if you have questions :smile:

**Comment by seltong** [comment]
ci:test:sf

**Comment by seltong** [comment]
ci:test:sf

**Comment by seltong** [comment]
ci:test:relevant

**Review by marcelabc — APPROVED** [review]
LGTM 🎉

**Comment by seltong** [comment]
Resent here: https://github.com/brianchandotcom/liferay-portal/pull/125308

**Pull request closed without merging**